device/globals/global_vm.gd

```diff
@@ -366,6 +366,9 @@ func get_object(name):
 	return objects[name]
 
 func register_object(name, val):
+	if !name:
+		report_errors("register_object", ["Name not given for " + val.get_class()])
```

Comment:
Can this happen? All of the calls to `register_object` seems to be guarded by `if global_id != ""` checks.

Reply:
It's more robust this way. I nearly passed in an empty `global_id` and someone else might in the future.

It's also a bit weird that you'd attach `target.gd` to a `Position2D` and wonder why it doesn't work, if you forgot to set the `global_id`. This is where the hardening is useful, to prevent bugs.

I would personally prefer this to be the only check for it and go so far as to remove the `if global_id != ""` checks altogether and receive errors instead. Errors are nice and clearly actionable.

Reply:
Yes, I agree. The call to `register_object` in `item.gd` should also be updated. The others (`telon.gd`, `bg_music.gd` and `player.gd`) hardcode the global_id, I think, so there's no need there.

Reply:
Turns out `bg_music` had it in the scene so I added it there for `telon` as well. Squashed into the commit that was about the item only.